daft/execution/execution_step.py

```diff
@@ -578,6 +579,26 @@ def _handle_file_write(self, input: MicroPartition) -> MicroPartition:
         )
 
 
+@dataclass(frozen=True)
+class CustomWrite(SingleOutputInstruction):
+    sink: DataSink
+
+    def run(self, inputs: list[MicroPartition]) -> list[MicroPartition]:
+        results = list(self.sink.write(iter(inputs)))
+        mp = MicroPartition.from_pydict({"custom_write_results": results})
```

Comment:
I think you want to force the input to be a python type here. so daft doesn't try to do any weird converting

Reply:
Gotcha, forced it to be a pyobj during conversion to Series